src/classes/UTIL_IntegrationConfig.cls

```diff
@@ -0,0 +1,184 @@
+/*
+    Copyright (c) 2019, Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group API
+* @description Config for external packages
+*/
+public abstract with sharing class UTIL_IntegrationConfig {
+    public Enum Integration {
+        ArchiveBridge
+    }
+
+    @TestVisible
+    private static final String DEFAULT_ARCHIVE_BRIDGE_NAMESPACE = 'dso_npsp';
```

Comment:
@seanpat09  There is a protected Custom Settings object (`Package_Settings__c`) that has the NPSP Namespace in it. I'm pretty sure this was added to allow for GEM testing with unmanaged NPSP code (though there is an `Advancement_Namespace__c` field on the `Contacts_Organizations_Settings__c` object). Could we add a field to the `Package_Settings__c` object for `DSO_Bridge_Namespace__c` to hold this value dynamically to allow for unmanaged code testing, or is that overkill for this because we could just modify the this class in an unmanaged context to change the namespace?

Reply:
I wanted to avoid adding another custom setting and figured that a combination of the `initNamespace` method and the `isInstalled` method could dynamically handle both the managed and unmanaged bridge installations.

* If the managed bridge package is installed, then `initNamespace` would use the hard coded namespace `dso_npsp` and `isInstalled` would return true

* If the unmanaged bridge package is installed, `initNamespace` would use a blank namespace since it won't find `dso__ArchiveRollupSummary__b`. Then `isInstalled` would check for a non-namespaced `ArchiveRollupSummary__b` object and for the `RollupType__c` field, find both and return true

* If it is not installed at all, then `isInstalled` shouldn't find the `ArchiveRollupSummary__b` at all and return false

So I think that covers handling both managed and unmanaged packages. The only thing it doesn't allow for is defining a custom namespace for the bridge package, but I don't think that's necessary, at least at this point.

Reply:
Ah. Nice implementation. That can work.